Adding to favorites crashes app

When I browse to any team or event, click the star button, check any of the boxes, then press the done button, the app will crash.

I am running version 4.3.0 of the app on Android 7.0.

When the app crashes, the Android Studio logger reports "[GoogleAccountDataServiceImpl] getToken() -> BAD_AUTHENTICATION"
It also gives this stack trace:

> E/UncaughtException: java.lang.RuntimeException: An error occurred while executing doInBackground()
                                                      at android.os.AsyncTask$3.done(AsyncTask.java:318)
                                                      at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:354)
                                                      at java.util.concurrent.FutureTask.setException(FutureTask.java:223)
                                                      at java.util.concurrent.FutureTask.run(FutureTask.java:242)
                                                      at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:243)
                                                      at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
                                                      at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
                                                      at java.lang.Thread.run(Thread.java:761)
                                                   Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'com.google.gson.JsonObject com.google.gson.JsonElement.getAsJsonObject()' on a null object reference
                                                      at com.thebluealliance.androidclient.datafeed.MyTbaDatafeed.updateModelSettings(MyTbaDatafeed.java:327)
                                                      at com.thebluealliance.androidclient.accounts.UpdateUserModelSettings.doInBackground(UpdateUserModelSettings.java:37)
                                                      at com.thebluealliance.androidclient.accounts.UpdateUserModelSettings.doInBackground(UpdateUserModelSettings.java:14)
                                                      at android.os.AsyncTask$2.call(AsyncTask.java:304)
                                                      at java.util.concurrent.FutureTask.run(FutureTask.java:237)
                                                      at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:243) 
                                                      at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133) 
                                                      at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607) 
                                                      at java.lang.Thread.run(Thread.java:761) 
03-26 18:18:24.428 527-662/? E/AndroidRuntime: FATAL EXCEPTION: AsyncTask #4
                                               Process: com.thebluealliance.androidclient, PID: 527
                                               java.lang.RuntimeException: An error occurred while executing doInBackground()
                                                   at android.os.AsyncTask$3.done(AsyncTask.java:318)
                                                   at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:354)
                                                   at java.util.concurrent.FutureTask.setException(FutureTask.java:223)
                                                   at java.util.concurrent.FutureTask.run(FutureTask.java:242)
                                                   at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:243)
                                                   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
                                                   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
                                                   at java.lang.Thread.run(Thread.java:761)
                                                Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'com.google.gson.JsonObject com.google.gson.JsonElement.getAsJsonObject()' on a null object reference
                                                   at com.thebluealliance.androidclient.datafeed.MyTbaDatafeed.updateModelSettings(MyTbaDatafeed.java:327)
                                                   at com.thebluealliance.androidclient.accounts.UpdateUserModelSettings.doInBackground(UpdateUserModelSettings.java:37)
                                                   at com.thebluealliance.androidclient.accounts.UpdateUserModelSettings.doInBackground(UpdateUserModelSettings.java:14)
                                                   at android.os.AsyncTask$2.call(AsyncTask.java:304)
                                                   at java.util.concurrent.FutureTask.run(FutureTask.java:237)
                                                   at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:243) 
                                                   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133) 
                                                   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607) 
                                                   at java.lang.Thread.run(Thread.java:761)`

--- a/android/src/main/java/com/thebluealliance/androidclient/activities/DatafeedActivity.java
+++ b/android/src/main/java/com/thebluealliance/androidclient/activities/DatafeedActivity.java
@@ -149,6 +149,7 @@ public abstract class DatafeedActivity extends BaseActivity
                     .datafeedModule(application.getDatafeedModule())
                     .binderModule(application.getBinderModule())
                     .databaseWriterModule(application.getDatabaseWriterModule())
+                    .authModule(application.getAuthModule())
                     .subscriberModule(new SubscriberModule(this))
                     .clickListenerModule(new ClickListenerModule(this))
                     .build();

--- a/android/src/main/java/com/thebluealliance/androidclient/activities/HomeActivity.java
+++ b/android/src/main/java/com/thebluealliance/androidclient/activities/HomeActivity.java
@@ -422,6 +422,7 @@ public class HomeActivity extends DatafeedActivity implements HasFragmentCompone
                     .binderModule(application.getBinderModule())
                     .databaseWriterModule(application.getDatabaseWriterModule())
                     .gceModule(application.getGceModule())
+                    .authModule(application.getAuthModule())
                     .subscriberModule(new SubscriberModule(this))
                     .clickListenerModule(new ClickListenerModule(this))
                     .build();

--- a/android/src/main/java/com/thebluealliance/androidclient/activities/TeamAtDistrictActivity.java
+++ b/android/src/main/java/com/thebluealliance/androidclient/activities/TeamAtDistrictActivity.java
@@ -166,6 +166,7 @@ public class TeamAtDistrictActivity extends DatafeedActivity
               .datafeedModule(application.getDatafeedModule())
               .binderModule(application.getBinderModule())
               .databaseWriterModule(application.getDatabaseWriterModule())
+              .authModule(application.getAuthModule())
               .subscriberModule(new SubscriberModule(this))
               .clickListenerModule(new ClickListenerModule(this))
               .build();

--- a/android/src/main/java/com/thebluealliance/androidclient/activities/TeamAtEventActivity.java
+++ b/android/src/main/java/com/thebluealliance/androidclient/activities/TeamAtEventActivity.java
@@ -208,6 +208,7 @@ public class TeamAtEventActivity extends MyTBASettingsActivity
               .datafeedModule(application.getDatafeedModule())
               .binderModule(application.getBinderModule())
               .databaseWriterModule(application.getDatabaseWriterModule())
+              .authModule(application.getAuthModule())
               .subscriberModule(new SubscriberModule(this))
               .clickListenerModule(new ClickListenerModule(this))
               .build();

--- a/android/src/main/java/com/thebluealliance/androidclient/activities/ViewDistrictActivity.java
+++ b/android/src/main/java/com/thebluealliance/androidclient/activities/ViewDistrictActivity.java
@@ -183,6 +183,7 @@ public class ViewDistrictActivity extends MyTBASettingsActivity
               .datafeedModule(application.getDatafeedModule())
               .binderModule(application.getBinderModule())
               .databaseWriterModule(application.getDatabaseWriterModule())
+              .authModule(application.getAuthModule())
               .subscriberModule(new SubscriberModule(this))
               .clickListenerModule(new ClickListenerModule(this))
               .build();

--- a/android/src/main/java/com/thebluealliance/androidclient/activities/ViewEventActivity.java
+++ b/android/src/main/java/com/thebluealliance/androidclient/activities/ViewEventActivity.java
@@ -267,6 +267,7 @@ public class ViewEventActivity extends MyTBASettingsActivity
                     .binderModule(application.getBinderModule())
                     .databaseWriterModule(application.getDatabaseWriterModule())
                     .gceModule(application.getGceModule())
+                    .authModule(application.getAuthModule())
                     .subscriberModule(new SubscriberModule(this))
                     .clickListenerModule(new ClickListenerModule(this))
                     .build();

--- a/android/src/main/java/com/thebluealliance/androidclient/activities/ViewMatchActivity.java
+++ b/android/src/main/java/com/thebluealliance/androidclient/activities/ViewMatchActivity.java
@@ -190,6 +190,7 @@ public class ViewMatchActivity extends MyTBASettingsActivity
               .datafeedModule(application.getDatafeedModule())
               .binderModule(application.getBinderModule())
               .databaseWriterModule(application.getDatabaseWriterModule())
+              .authModule(application.getAuthModule())
               .subscriberModule(new SubscriberModule(this))
               .clickListenerModule(new ClickListenerModule(this))
               .build();

--- a/android/src/main/java/com/thebluealliance/androidclient/activities/ViewTeamActivity.java
+++ b/android/src/main/java/com/thebluealliance/androidclient/activities/ViewTeamActivity.java
@@ -524,6 +524,7 @@ public class ViewTeamActivity extends MyTBASettingsActivity implements
                     .binderModule(application.getBinderModule())
                     .databaseWriterModule(application.getDatabaseWriterModule())
                     .gceModule(application.getGceModule())
+                    .authModule(application.getAuthModule())
                     .subscriberModule(new SubscriberModule(this))
                     .clickListenerModule(new ClickListenerModule(this))
                     .build();

--- a/android/src/main/java/com/thebluealliance/androidclient/auth/firebase/FirebaseAuthProvider.java
+++ b/android/src/main/java/com/thebluealliance/androidclient/auth/firebase/FirebaseAuthProvider.java
@@ -2,10 +2,12 @@ package com.thebluealliance.androidclient.auth.firebase;
 
 import android.content.Intent;
 
+import com.google.android.gms.tasks.Task;
 import com.google.firebase.auth.AuthCredential;
 import com.google.firebase.auth.AuthResult;
 import com.google.firebase.auth.FirebaseAuth;
 import com.google.firebase.auth.FirebaseUser;
+import com.google.firebase.auth.GetTokenResult;
 import com.thebluealliance.androidclient.TbaLogger;
 import com.thebluealliance.androidclient.auth.AuthProvider;
 import com.thebluealliance.androidclient.auth.User;
@@ -57,6 +59,19 @@ public class FirebaseAuthProvider implements AuthProvider {
         }
 
         return new FirebaseSignInUser(firebaseUser);
+    }
+
+    @Nullable
+    public Task<GetTokenResult> getUserToken(boolean forceRefresh) {
+        if (mFirebaseAuth == null) {
+            return null;
+        }
+        FirebaseUser firebaseUser = mFirebaseAuth.getCurrentUser();
+        if (firebaseUser == null) {
+            return null;
+        }
+
+        return firebaseUser.getIdToken(forceRefresh);
     }
 
     @Nullable @Override

--- a/android/src/main/java/com/thebluealliance/androidclient/datafeed/APIv3RequestInterceptor.java
+++ b/android/src/main/java/com/thebluealliance/androidclient/datafeed/APIv3RequestInterceptor.java
@@ -41,6 +41,7 @@ public class APIv3RequestInterceptor implements Interceptor {
         Request originalRequest = chain.request();
         String url = originalRequest.url().toString();
         if (!url.contains("thebluealliance.com/api/v3")) {
+            TbaLogger.d("FETCHING " + url);
             return chain.proceed(originalRequest);
         }
 

--- a/android/src/main/java/com/thebluealliance/androidclient/datafeed/MyTbaDatafeed.java
+++ b/android/src/main/java/com/thebluealliance/androidclient/datafeed/MyTbaDatafeed.java
@@ -309,13 +309,18 @@ public class MyTbaDatafeed {
                 Response<ModelsMobileApiMessagesBaseResponse> response = mModelApi
                         .setPreferences(authHeader, request)
                         .execute();
-                if (response == null || !response.isSuccessful()) {
+                if (response == null) {
+                    TbaLogger.w("Null response for mytba update");
+                    return ModelPrefsResult.ERROR;
+                }
+                if (!response.isSuccessful()) {
+                    TbaLogger.w("mytba update error " + response.code() + ": " + response.message());
                     return ModelPrefsResult.ERROR;
                 }
 
                 ModelsMobileApiMessagesBaseResponse prefResponse = response.body();
-                TbaLogger.d("Result: " + prefResponse.code + "/" + prefResponse.message);
-                if (response.code() == 401) {
+                TbaLogger.d("Mytba result: " + prefResponse.code + "/" + prefResponse.message);
+                if (response.code() == 401 || prefResponse.code == 401) {
                     TbaLogger.e(prefResponse.message);
                     return ModelPrefsResult.ERROR;
                 }

--- a/android/src/main/java/com/thebluealliance/androidclient/datafeed/gce/GceAuthController.java
+++ b/android/src/main/java/com/thebluealliance/androidclient/datafeed/gce/GceAuthController.java
@@ -8,10 +8,16 @@ import android.support.annotation.WorkerThread;
 
 import com.google.android.gms.auth.GoogleAuthException;
 import com.google.android.gms.auth.GoogleAuthUtil;
+import com.google.android.gms.tasks.Task;
+import com.google.android.gms.tasks.Tasks;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.GetTokenResult;
 import com.thebluealliance.androidclient.TbaLogger;
 import com.thebluealliance.androidclient.accounts.AccountController;
+import com.thebluealliance.androidclient.auth.firebase.FirebaseAuthProvider;
 
 import java.io.IOException;
+import java.util.concurrent.ExecutionException;
 
 import javax.inject.Inject;
 
@@ -21,93 +27,38 @@ import javax.inject.Inject;
  * Resulting oauth tokens are to be used with the GCE Retrofit Services
  */
 public class GceAuthController {
-    private static final int MAX_BACKOFF_TRIES = 3;
     private static final String AUTH_HEADER_FORMAT = "Bearer %1$s";
 
-    private final Context mContext;
-    private final AccountController mAccountController;
-
-    /* Time to wait during exponential backoff (ms) */
-    private long mBackoffTime;
-
-    /* Number of tries we've made while getting token */
-    private int mBackoffCount;
+    private final FirebaseAuthProvider mFirebaseAuth;
 
     @Inject
-    public GceAuthController(
-            Context context,
-            AccountController accountController) {
-        mContext = context;
-        mAccountController = accountController;
-        resetBackoff();
-    }
-
-    private void resetBackoff() {
-        mBackoffTime = 1000;
-        mBackoffCount = 0;
-    }
-
-    /**
-     * Negotiates a Google oauth2 token to use with Cloud Endpoints + Retrofit
-     * This <b>MUST</b> be run from a background thread, performs much network operations and sleep
-     * Based on https://developers.google.com/api-client-library/java/google-api-java-client/reference/1.19.1/com/google/api/client/googleapis/extensions/android/gms/auth/GoogleAccountCredential
-     * Specifically #getToken()
-     * @return An auth token to be used with Cloud Endpoints
-     * @throws GoogleAuthException Trouble authenticating to Google
-     */
-    @WorkerThread @VisibleForTesting
-    @Nullable private String getAuthTokenWithBackoff() throws GoogleAuthException {
-        String scope = getAudience();
-        String account = mAccountController.getSelectedAccount();
-        if (account == null || account.isEmpty()) {
-            TbaLogger.e("No system account found, can't get auth token");
-            return null;
-        }
-        resetBackoff();
-        while (mBackoffCount < MAX_BACKOFF_TRIES) {
-            try {
-                return getGoogleAuthToken(account, scope);
-            } catch (IOException e) {
-                TbaLogger.i("Unable to get token, sleeping " + mBackoffTime + " ms");
-                e.printStackTrace();
-                SystemClock.sleep(mBackoffTime);
-                mBackoffTime *= 2;
-                mBackoffCount++;
-            }
-        }
-        return null;
+    public GceAuthController(FirebaseAuthProvider firebaseAuth) {
+        mFirebaseAuth = firebaseAuth;
     }
 
     /**
      * Builds the correct Authorization header to be used with GCE requests
-     * This method <b>MUST</b> be called from a background thread; it uses {@link #getAuthTokenWithBackoff()}
+     * This method <b>MUST</b> be called from a background thread; it will block on a Future
      * @return Authorization header, or null if {@link GoogleAuthException} or other error happened
      */
     @WorkerThread
     public @Nullable String getAuthHeader() {
         try {
-            String token = getAuthTokenWithBackoff();
+            Task<GetTokenResult> tokenTask = mFirebaseAuth.getUserToken(false);
+            if (tokenTask == null) {
+                TbaLogger.i("No firebase user found, can't fetch auth token");
+                return null;
+            }
+            GetTokenResult tokenResult = Tasks.await(tokenTask);
+            String token = tokenResult.getToken();
             if (token == null) {
                 return null;
             }
             return String.format(AUTH_HEADER_FORMAT, token);
-        } catch (GoogleAuthException e) {
-            TbaLogger.w("Auth exception while fetching google token");
+        } catch (InterruptedException | ExecutionException e) {
+            TbaLogger.w("Auth exception while fetching google token", e);
             return null;
         }
-
-    }
-
-    @WorkerThread @VisibleForTesting
-    String getGoogleAuthToken(String account, String scope)
-    throws IOException, GoogleAuthException {
-        if (account == null || account.isEmpty()) return null;
-        return GoogleAuthUtil.getToken(mContext, account, scope);
-    }
-
-    private String getAudience() {
-        String webClientId = mAccountController.getWebClientId();
-        return "audience:server:client_id:" + webClientId;
     }
 
 }

--- a/android/src/main/java/com/thebluealliance/androidclient/datafeed/gce/GceAuthController.java
+++ b/android/src/main/java/com/thebluealliance/androidclient/datafeed/gce/GceAuthController.java
@@ -44,13 +44,7 @@ public class GceAuthController {
     @WorkerThread
     public @Nullable String getAuthHeader() {
         try {
-            Task<GetTokenResult> tokenTask = mFirebaseAuth.getUserToken(false);
-            if (tokenTask == null) {
-                TbaLogger.i("No firebase user found, can't fetch auth token");
-                return null;
-            }
-            GetTokenResult tokenResult = Tasks.await(tokenTask);
-            String token = tokenResult.getToken();
+            String token = getGoogleAuthToken();
             if (token == null) {
                 return null;
             }
@@ -59,6 +53,17 @@ public class GceAuthController {
             TbaLogger.w("Auth exception while fetching google token", e);
             return null;
         }
+    }
+
+    @WorkerThread @VisibleForTesting @Nullable
+    String getGoogleAuthToken() throws ExecutionException, InterruptedException {
+        Task<GetTokenResult> tokenTask = mFirebaseAuth.getUserToken(false);
+        if (tokenTask == null) {
+            TbaLogger.i("No firebase user found, can't fetch auth token");
+            return null;
+        }
+        GetTokenResult tokenResult = Tasks.await(tokenTask);
+        return tokenResult.getToken();
     }
 
 }

--- a/android/src/main/java/com/thebluealliance/androidclient/datafeed/gce/GceAuthController.java
+++ b/android/src/main/java/com/thebluealliance/androidclient/datafeed/gce/GceAuthController.java
@@ -1,7 +1,5 @@
 package com.thebluealliance.androidclient.datafeed.gce;
 
-import android.content.Context;
-import android.os.SystemClock;
 import android.support.annotation.Nullable;
 import android.support.annotation.VisibleForTesting;
 import android.support.annotation.WorkerThread;
@@ -10,13 +8,10 @@ import com.google.android.gms.auth.GoogleAuthException;
 import com.google.android.gms.auth.GoogleAuthUtil;
 import com.google.android.gms.tasks.Task;
 import com.google.android.gms.tasks.Tasks;
-import com.google.firebase.auth.FirebaseAuth;
 import com.google.firebase.auth.GetTokenResult;
 import com.thebluealliance.androidclient.TbaLogger;
-import com.thebluealliance.androidclient.accounts.AccountController;
 import com.thebluealliance.androidclient.auth.firebase.FirebaseAuthProvider;
 
-import java.io.IOException;
 import java.util.concurrent.ExecutionException;
 
 import javax.inject.Inject;

--- a/android/src/main/java/com/thebluealliance/androidclient/datafeed/gce/GceModule.java
+++ b/android/src/main/java/com/thebluealliance/androidclient/datafeed/gce/GceModule.java
@@ -11,6 +11,9 @@ import com.appspot.tbatv_prod_hrd.TeamMedia;
 import com.google.gson.Gson;
 import com.thebluealliance.androidclient.accounts.AccountController;
 import com.thebluealliance.androidclient.accounts.AccountModule;
+import com.thebluealliance.androidclient.auth.AuthModule;
+import com.thebluealliance.androidclient.auth.AuthProvider;
+import com.thebluealliance.androidclient.auth.firebase.FirebaseAuthProvider;
 import com.thebluealliance.androidclient.config.AppConfig;
 import com.thebluealliance.androidclient.database.Database;
 import com.thebluealliance.androidclient.datafeed.HttpModule;
@@ -30,7 +33,7 @@ import retrofit2.Retrofit;
 /**
  * Dagger Module for Google Cloud Endpoints and their related things
  */
-@Module(includes = {HttpModule.class, AccountModule.class, GcmModule.class})
+@Module(includes = {HttpModule.class, AccountModule.class, GcmModule.class, AuthModule.class})
 public class GceModule {
 
     // Format with app engine project ID
@@ -46,9 +49,8 @@ public class GceModule {
     }
 
     @Provides GceAuthController provideGceAuthController(
-            Context context,
-            AccountController accountController) {
-        return new GceAuthController(context, accountController);
+            @Named("firebase_auth") AuthProvider firebaseAuthProvider) {
+        return new GceAuthController((FirebaseAuthProvider)firebaseAuthProvider);
     }
 
     @Provides @Singleton @Named("gce_retrofit")

--- a/android/src/test/java/com/thebluealliance/androidclient/datafeed/gce/GceAuthControllerTest.java
+++ b/android/src/test/java/com/thebluealliance/androidclient/datafeed/gce/GceAuthControllerTest.java
@@ -5,6 +5,7 @@ import android.content.Context;
 import com.google.android.gms.auth.GoogleAuthException;
 import com.thebluealliance.androidclient.DefaultTestRunner;
 import com.thebluealliance.androidclient.accounts.AccountController;
+import com.thebluealliance.androidclient.auth.firebase.FirebaseAuthProvider;
 
 import org.junit.Before;
 import org.junit.Test;
@@ -13,6 +14,7 @@ import org.mockito.Mock;
 import org.mockito.MockitoAnnotations;
 
 import java.io.IOException;
+import java.util.concurrent.ExecutionException;
 
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNull;
@@ -24,34 +26,32 @@ import static org.mockito.Mockito.when;
 public class GceAuthControllerTest {
 
     private static final String AUTH_TOKEN = "abc123";
-    private static final String CLIENT_ID = "foo.bar";
-    private static final String ACCOUNT = "foo@bar.com";
 
-    @Mock Context mContext;
-    @Mock AccountController mAccountController;
+    @Mock FirebaseAuthProvider mFirebaseAuth;
 
     private GceAuthController mAuthController;
 
     @Before
-    public void setUp() throws GoogleAuthException, IOException {
+    public void setUp() throws ExecutionException, InterruptedException {
         MockitoAnnotations.initMocks(this);
-        mAuthController = spy(new GceAuthController(mContext, mAccountController));
+        mAuthController = spy(new GceAuthController(mFirebaseAuth));
 
         doReturn(AUTH_TOKEN)
                 .when(mAuthController)
-                .getGoogleAuthToken(ACCOUNT, "audience:server:client_id:foo.bar");
+                .getGoogleAuthToken();
     }
 
     @Test
     public void testGetAuthHeader() {
-        when(mAccountController.getSelectedAccount()).thenReturn(ACCOUNT);
-        when(mAccountController.getWebClientId()).thenReturn(CLIENT_ID);
         String header = mAuthController.getAuthHeader();
         assertEquals(header, "Bearer abc123");
     }
 
     @Test
-    public void testGetAuthHeaderNoAccount() {
+    public void testGetAuthHeaderNoAccount() throws ExecutionException, InterruptedException {
+        doReturn(null)
+                .when(mAuthController)
+                .getGoogleAuthToken();
         String header = mAuthController.getAuthHeader();
         assertNull(header);
     }

--- a/android/src/test/java/com/thebluealliance/androidclient/datafeed/gce/GceAuthControllerTest.java
+++ b/android/src/test/java/com/thebluealliance/androidclient/datafeed/gce/GceAuthControllerTest.java
@@ -1,10 +1,6 @@
 package com.thebluealliance.androidclient.datafeed.gce;
 
-import android.content.Context;
-
-import com.google.android.gms.auth.GoogleAuthException;
 import com.thebluealliance.androidclient.DefaultTestRunner;
-import com.thebluealliance.androidclient.accounts.AccountController;
 import com.thebluealliance.androidclient.auth.firebase.FirebaseAuthProvider;
 
 import org.junit.Before;
@@ -13,14 +9,12 @@ import org.junit.runner.RunWith;
 import org.mockito.Mock;
 import org.mockito.MockitoAnnotations;
 
-import java.io.IOException;
 import java.util.concurrent.ExecutionException;
 
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNull;
 import static org.mockito.Mockito.doReturn;
 import static org.mockito.Mockito.spy;
-import static org.mockito.Mockito.when;
 
 @RunWith(DefaultTestRunner.class)
 public class GceAuthControllerTest {

